File: Sowndariya_Assessment24/Pages/searchPage.py
from selenium.webdriver.common.by import By


class SearchPage:

    def __init__(self, driver):
        self.driver = driver

    def search_box(self):
        return self.driver.find_element(By.NAME,"search")

    def search_button(self):
        return self.driver.find_element(By.CSS_SELECTOR,"#search button")

    def product_names(self):
        return self.driver.find_elements(By.XPATH,"//div[@class='product-thumb']//h4/a")

File: Sowndariya_Assesssment03/TechnicalAssessment03/src/test/java/com/pages/SearchPage.java
package com.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class SearchPage {
    WebDriver driver;

    public SearchPage(WebDriver driver){
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }
    @FindBy(name = "search")
    WebElement searchBox;

    @FindBy(xpath = "//button[@class='btn btn-default btn-lg']")
    WebElement searchButton;

    public void enterSearchKeyword(String keyword){
        searchBox.sendKeys(keyword);
    }

    public void clickSearchButton(){
        searchButton.click();
    }

    public boolean noProductMessageDisplayed(){
        return driver.getPageSource().contains("There is no product that matches the search criteria.");
    }
}